torch/distributed/checkpoint/planner_helpers.py

```diff
@@ -185,8 +188,40 @@ def _create_default_metadata_only_plan(state_dict: STATE_DICT_TYPE) -> SavePlan:
     return SavePlan(requests)
 
 
+def _get_box_for(
+    tensor: DTensor, idx: Optional[int]
+) -> Tuple[torch.Size, torch.Size]:
+    device_mesh = tensor.device_mesh
+    assert device_mesh.ndim == 1, "Only 1D DeviceMeshes currently handled"
+
+    size = tensor.to_local().size()
+    offsets = tensor._spec.local_offsets
+
+    return (torch.Size(offsets), size)
+
+
+def _create_write_items_for_dtensor(fqn: str, tensor: DTensor) -> WriteItem:
+    offsets, sizes = _get_box_for(
+        tensor, tensor.device_mesh.get_coordinate_on_dim(0)
+    )
+    return WriteItem(
+        index=MetadataIndex(fqn, offsets),
+        type=WriteItemType.SHARD,
```

Comment:
what's this WriteItemType.SHARD? DTensor have both shard/replicate, would this writeItemType still meaningful here?

Reply:
Ye. Will file an issue for this. This part will need to be redesigned and updated when we move away from shardedtensor.

Reply:
> what's this WriteItemType.SHARD? DTensor have both shard/replicate, would this writeItemType still meaningful here?

There are currently three WriteItemType, Byte, Shard, and Tensor.  DTensor can use the shard type for now, and we can clean up the naming when we are ready to deprecate ST from DCP.